pkg/domain/domain.go

```diff
@@ -271,7 +271,11 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	// 4. No regenrated schema diff.
 	startTime := time.Now()
 	if currentSchemaVersion != 0 && neededSchemaVersion > currentSchemaVersion && neededSchemaVersion-currentSchemaVersion < LoadSchemaDiffVersionGapThreshold {
-		is, relatedChanges, diffTypes, err := do.tryLoadSchemaDiffs(m, currentSchemaVersion, neededSchemaVersion, schemaTs)
+		schemaTsOrStartTs := schemaTs
+		if schemaTsOrStartTs == 0 {
+			schemaTsOrStartTs = startTS
```

Comment:
Could set it before line242？ Then we needn't do this check.

Reply:
`getTimestampForSchemaVersionWithNonEmptyDiff` always return schemaTs=0 ... (something is wrong but all the test cases still pass)
**I'm not sure using startTs as schemaTs for infoschema v1 is correct or not, so I put the code here.
This way it only affect infoscheme v2.**

Reply:
We also can use `schemaTsOrStartTs=startTs` in line242. So the `schemaTsOrStartTs` only affect v2?

Reply:
![Uploading 9fWRb3yua4.jpg…]()

What I'm saying is that you can move this line over here, so that it doesn't affect the original logic and it does what you want